tools/driver/swift_format_main.cpp

```diff
@@ -0,0 +1,295 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  Formats Swift files or file ranges according to a set of parameters.
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/Format/Format.h"
+#include "swift/AST/DiagnosticsFrontend.h"
+#include "swift/Basic/SourceManager.h"
+#include "swift/Frontend/Frontend.h"
+#include "swift/Frontend/PrintingDiagnosticConsumer.h"
+#include "swift/IDE/Formatting.h"
+#include "swift/Option/Options.h"
+#include "swift/Subsystems.h"
+#include "llvm/Option/Arg.h"
+#include "llvm/Option/Option.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+
+#include <string>
+#include <vector>
+
+using namespace swift;
+using namespace swift::ide;
+using namespace llvm::opt;
+
+class FormatterDocument {
+private:
+  SourceManager SM;
+  unsigned BufferID;
+  CompilerInvocation CompInv;
+  std::unique_ptr<ParserUnit> Parser;
+  class FormatterDiagConsumer: public swift::DiagnosticConsumer {
+    void handleDiagnostic(SourceManager &SM, SourceLoc Loc,
+                          DiagnosticKind Kind, StringRef Text,
+                          const swift::DiagnosticInfo &Info) override {
+      llvm::errs() << "Parse error: " << Text << "\n";
+    }
+  } DiagConsumer;
+
+public:
+  FormatterDocument(std::unique_ptr<llvm::MemoryBuffer> Buffer) {
+    updateCode(std::move(Buffer));
+  }
+
+  void updateCode(std::unique_ptr<llvm::MemoryBuffer> Buffer) {
+    BufferID = SM.addNewSourceBuffer(std::move(Buffer));
+    Parser.reset(new ParserUnit(SM, BufferID, CompInv.getLangOptions(),
+                                CompInv.getModuleName()));
+    Parser->getDiagnosticEngine().addConsumer(DiagConsumer);
+    auto &P = Parser->getParser();
+    for (bool Done = false; !Done; Done = P.Tok.is(tok::eof)) {
+      P.parseTopLevel();
+    }
+  }
+
+  std::pair<LineRange, std::string> reformat(LineRange Range,
+                                             CodeFormatOptions Options) {
+    return ::reformat(Range,
+                      Options,
+                      SM,
+                      Parser->getSourceFile());
+  }
+
+  const llvm::MemoryBuffer &memBuffer() const {
+    return *SM.getLLVMSourceMgr().getMemoryBuffer(BufferID);
+  }
+};
+
+class SwiftFormatInvocation {
+private:
+  std::string MainExecutablePath;
+  std::string OutputFilename = "-";
+  std::vector<std::string> InputFilenames;
+  bool UseTabs = false;
+  bool InPlace = false;
+  unsigned TabWidth = 4;
+  unsigned IndentWidth = 4;
```

Comment:
I've taken the defaults from the `CodeFormatOptions` struct in `Formatting.h`. In a future PR I plan to extract that into a "style" structure so that it is not duplicated across the codebase.